packages/app/components/document/document-header.tsx

```diff
@@ -9,30 +13,112 @@ import ModelIcon from '../model-icon'
 import styles from './document-header.module.css'
 import DocumentStateBadge from './document-state-badge'
 
+type PropsType = {
+    activePanel?: string
+    document?: Document
+    isJsonPanelOpen?: boolean
+    model?: ModelWithWorkflow
+    version?: any
+    toggleJsonDiffer?: () => void
+    togglePanelOpen?: (panel: string) => void
+    privileges?: string[]
+    comments?: DocumentComment[]
+    history?: DocumentHistory[]
+    collaborators?: Collaborator[]
+}
+
 const DocumentHeader = ({
     activePanel = null,
     document = null,
     isJsonPanelOpen = false,
     model,
     version = null,
-    toggleJsonDiffer,
-    togglePanelOpen,
+    toggleJsonDiffer = () => {},
+    togglePanelOpen = () => {},
     privileges = [],
     comments = [],
     history = [],
-}) => {
+    collaborators = [],
+}: PropsType) => {
     const numberOfComments =
         comments === null ? 0 : comments.filter(c => !c.resolved).length
     const numberOfHistoryEntries = history === null ? 0 : history.length
+    const sortedCollaborators = [...collaborators]
+        .sort(({ hasBeenActive }) => (hasBeenActive ? -1 : 0))
+        .sort(({ isActive }) => (isActive ? -1 : 0))
+    const priorityCollaborators = sortedCollaborators.slice(0, 4)
+    const plusCollaborators = sortedCollaborators.slice(4)
 
     return (
-        <div>
-            <div className={styles.title}>
-                <ModelIcon name={model?.icon?.name} color={model?.icon?.color} />
-                {model?.name}
+        <div className={styles.header}>
+            <div className={styles.titleDescription}>
+                <div className={styles.title}>
+                    <ModelIcon name={model?.icon?.name} color={model?.icon?.color} />
+                    <h2 className="h3">{model?.name}</h2>
+                </div>
+
+                <div className={styles.description}>{model?.description}</div>
             </div>
 
-            <div className={styles.description}>{model?.description}</div>
+            <h3 className="sr-only">Collaborators</h3>
+            <div className={styles.collaborators}>
```

Comment:
I generally prefer slightly longer `.tsx` files to more components, but at some point, that line is crossed. Please let me know if you think this should be its own component.